controllers/selfnoderemediation_controller.go

```diff
@@ -624,3 +630,28 @@ func (r *SelfNodeRemediationReconciler) restoreNode(nodeToRestore *v1.Node) (ctr
 	// all done, stop reconciling
 	return ctrl.Result{Requeue: true}, nil
 }
+
+func (r *SelfNodeRemediationReconciler) updateSnrStatusLastError(snr *v1alpha1.SelfNodeRemediation, err error) error {
+	var lastErrorVal string
+
+	if err != nil {
+		lastErrorVal = err.Error()
+	} else {
+		lastErrorVal = "successful reconcile"
+	}
+
+	if snr.Status.LastError != lastErrorVal {
+		snr.Status.LastError = lastErrorVal
+		updateErr := r.Client.Status().Update(context.Background(), snr)
+		if updateErr != nil {
+			r.logger.Error(updateErr, "Failed to update SelfNodeRemediation status")
```

Comment:
We don't want to retry when this fails?
Also, because every PP agent is potentially trying to update the CR, I think that there is a good chance that `Update` fails. I think `Patch` would be better here.

Reply:
will change to `Patch` :)
About retrying when this fails, I asked Michael and we decided not to because we thought we don't want to add another reason for the reconcile to fail (IIUC). Wdyt?  @mshitrit

Reply:
IIRC we thought it does not make sense to return an error for fail  because it'll overshadow the original error.
However Marc brings up a good point - I think we can retry without returning an error that will overshadow the original error.
However since we are a bit short on time I suggest to proceed as is and just open a Jira ticket to capture this.

Reply:
Sorry, saw the comments only just now.
I don't think we overshadow anything.
We just reconcile again. We run into the same error. Then we try to update the status again. Until it succeeds.

Reply:
>I don't think we overshadow anything.

My intention is that now instead of returning the initial error that caused the malfunction, we will return an error that say that we couldn't updated the status.